### PR TITLE
Keybind UI utils

### PR DESCRIPTION
**Pull request by MrStashley**
This pr creates utils for displaying keybindings, including the ability to "pretty print" them, or show their respective keyboard image.

I also created an "info" section that can be added to a keybind list and can be queried. I think in the future this will allow us to get quick tool tips for each of the key binds, and also maybe in the future we could give chatGPT key bind info and allow it to give smart suggestions

### CODE REVIEW

File: src/util/keyutil.ts [hunk 1]
```diff
@@ -114,6 +123,79 @@ class KeybindManager {
         this.keyDescriptionsMap = newKeyDescriptions;
     }
 
+    prettyPrintKeybind(keyDescription: string): string {
+        let keyPress = parseKeyDescription(keyDescription);
+        let returnString = "";
+        if (keyPress.mods.Cmd) {
+            returnString += "⌘";
```
Comment (MrStashley) [inline src/util/keyutil.ts:130]: should I make these platform specific? I was wondering if I should make it the windows key on linux, idk

File: src/util/keyutil.ts [hunk 2]
```diff
@@ -114,6 +123,79 @@ class KeybindManager {
...
+        if (keyPress.mods.Ctrl) {
+            returnString += "⌃";
+        }
+        if (keyPress.mods.Option) {
+            returnString += "⌥";
+        }
+        if (keyPress.mods.Shift) {
+            returnString += "⇧";
+        }
+        if (keyPress.mods.Meta) {
```
Comment (MrStashley) [inline src/util/keyutil.ts:141]: How the system currently works is Cmd is automatically MetaKey on Mac and AltKey on linux and option is the opposite. But if a Keybind specifies meta it's always looking for the meta key. kind of like how you can use key code to specify a physical key, I currently have it so meta and alt allow someone to override any platform specific behavior. Does that make sense? It's definitely a niche use case but I figured there's no reason not to have it

Comment (MrStashley) [reply src/util/keyutil.ts:141]: Also, in this case the other key would be ignored, so like if for some reason there is a keybinding that is `Meta:Cmd:c` that is interpreted as `Meta:c`, and `Alt:Option:c` is interpreted as `Option:c`. I feel like that is the most reasonable way to interpret those cases anyways

File: src/util/keyutil.ts (hunk 1 above)
Comment (sawka) [reply src/util/keyutil.ts:130]: ya, that's a good idea to make them platform specific.

File: src/util/keyutil.ts [hunk 3]
```diff
@@ -25,7 +25,8 @@ const KeyTypeCode = "code";
 
 type KeybindCallback = (event: WaveKeyboardEvent) => boolean;
 type KeybindConfigArray = Array<KeybindConfig>;
-type KeybindConfig = { command: string; keys: Array<string>; commandStr?: string };
+type KeybindConfig = { command: string; keys: Array<string>; commandStr?: string; info?: string };
+type KeybindUIDescription = { keys: Array<string>; info: string };
```
Comment (sawka) [inline src/util/keyutil.ts:29]: add "command" into this as well.  the command name carries a lot of information and is useful even without the info sometimes.

File: src/models/model.ts [hunk 4]
```diff
@@ -206,6 +206,7 @@ class Model {
             return resp.json();
         }).then((userKeybindings) => {
             this.keybindManager.setUserKeybindings(userKeybindings);
+            console.log("all keybindings:", this.keybindManager.getAllKeybindUIDescriptions());
```
Comment (sawka) [inline src/models/model.ts:209]: remove console.log.  we can see this by just accessing it directly in the console if we want it. `GlobalModel.keybindManager.getAllKeybindUIDescriptions()`

**Pull request merged**